sycl/include/sycl/esimd/detail/esimd_host_util.hpp

```diff
@@ -0,0 +1,362 @@
+//==----------------- esimd_host_util.hpp - DPC++ Explicit SIMD API --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Utility functions and definitions for implementing ESIMD intrinsics on host
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef __SYCL_DEVICE_ONLY__
+
+#define SIMDCF_ELEMENT_SKIP(i)
+
+namespace cl {
+namespace sycl {
+namespace detail {
+namespace half_impl {
+class half;
+} // namespace half_impl
+} // namespace detail
+} // namespace sycl
+} // namespace cl
+
+using half = cl::sycl::detail::half_impl::half;
+
+#define SAT 1
+
+namespace EsimdEmulSys {
+
+static long long abs(long long a) {
+  if (a < 0) {
+    return -a;
+  } else {
+    return a;
+  }
+}
+
+template <typename RT> struct satur {
+  template <typename T> static RT saturate(const T val, const int flags) {
+    if ((flags & SAT) == 0) {
+      return (RT)val;
+    }
+
+#ifdef max
+#undef max
+#undef min
```

Comment:
And what about the case `min` is a macro but not `max`? :-)
Anyway, trying to survive to random macros seems like a challenge... :-(

Reply:
resolved